src/app/modules/library/personal-library/personal-library.component.ts

```diff
@@ -1,21 +1,46 @@
-import { Component, Inject } from '@angular/core';
+import { Component, Inject, Signal, WritableSignal, computed, signal } from '@angular/core';
 import { LibraryProject } from '../libraryProject';
 import { LibraryService } from '../../../services/library.service';
 import { LibraryComponent } from '../library/library.component';
 import { MatDialogRef, MAT_DIALOG_DATA, MatDialog } from '@angular/material/dialog';
+import { ProjectFilterValues } from '../../../domain/projectFilterValues';
+import { ArchiveProjectService } from '../../../services/archive-project.service';
+import { Subscription } from 'rxjs';
+import { PageEvent } from '@angular/material/paginator';
 
 @Component({
   selector: 'app-personal-library',
   templateUrl: './personal-library.component.html',
   styleUrls: ['./personal-library.component.scss']
 })
 export class PersonalLibraryComponent extends LibraryComponent {
-  projects: LibraryProject[] = [];
   filteredProjects: LibraryProject[] = [];
-  personalProjects: LibraryProject[] = [];
-  sharedProjects: LibraryProject[] = [];
+  protected personalProjects: LibraryProject[] = [];
+  projects: LibraryProject[] = [];
+  protected selectedProjects: WritableSignal<LibraryProject[]> = signal([]);
+  protected sharedProjects: LibraryProject[] = [];
+  protected showArchived: boolean = false;
 
-  constructor(protected dialog: MatDialog, protected libraryService: LibraryService) {
+  protected numSelectedProjects: Signal<number> = computed(() => this.selectedProjects().length);
+  protected projectIdToIsSelected: Signal<any> = computed(() =>
```

Comment:
Give type to signal?
```suggestion
  protected projectIdToIsSelected: Signal<{ [key: number]: boolean }> = computed(() =>
```